TACC/Core-Styles
# feat/tup 617: Migrate c-card fixes to core-styles

## Overview

Migrate c-card style fixes to core-styles

<details><summary>Archived Notices about Button a Card</summary>

> **Warning**
> Should not be merged until the code is adapted to not conflict with Bootstrap .card in any Portal (both TUP and Core Portal).

> **Note**
> Following code is almost identical, but allows for more specificity. Duplicated/Identical code can be found on line 174-176
> ```
> /* To more specifically control c-card link font weight */
> :is(.card, :--c-card) a:--c-button {
>     font-weight: var(--bold);
> }
> ```

> **Important**
> Matt S. mentioned that they prefer the bold text for these buttons, but could use some letter spacing. It seems to me that this is a specificity issue and seems to be overrun by 
> 
> https://github.com/TACC/Core-Styles/blob/b20884a012e50bc33c79aa077a3824ca172ba1ab/src/lib/_imports/components/c-button.css#L35
> 
> ```
> :--c-button:where(:not(:--c-button--as-link)) {
>   font-weight: var(--medium);
> }
> ```

</details>

## Related

- [TUP-617](https://jira.tacc.utexas.edu/browse/TUP-617)
- Migrate from [c-card](https://github.com/TACC/tup-ui/blob/main/apps/tup-cms/src/taccsite_custom/tup_cms/static/tup_cms/css/for-core-styles/components/c-card.css)

## Changes

Importing c-card styles from tup-ui to core-styles

## Testing

- Check [here](https://www.tacc.utexas.edu/education/k-12-students/high-school-camps/gencyber/) Application card for button link styles
- Check [here](https://www.tacc.utexas.edu/about/staff-directory/) for staff image cards (specifically h4, and last paragraph margin-bottoms)

## UI

| BEFORE | AFTER |
| --- | --- |
| <img width="2560" alt="Screenshot 2023-10-26 at 2 20 46 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/4696afe1-5158-49f2-a617-285a9a6aa377"> | <img width="2408" alt="Screenshot 2023-10-26 at 4 10 54 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/f119b869-395c-4fc8-938a-62919afd9ff7"> |
| <img width="2560" alt="Screenshot 2023-10-26 at 2 22 03 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/d1188111-1318-47bb-980a-bf307759a9c2"> | <img width="2560" alt="Screenshot 2023-10-26 at 3 32 14 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/0dfa9fc4-0fb0-4c6b-9ad8-dc871d2f06c6"> |
| <img width="2560" alt="Screenshot 2023-10-26 at 2 23 19 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/71315f98-9701-43a8-9079-14241d482d7b"> | <img width="2560" alt="Screenshot 2023-10-26 at 3 31 28 PM" src="https://github.com/TACC/Core-Styles/assets/63771558/c8e6a1ff-dce4-4456-9650-cbff06c9fe00"> |
